cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -413,7 +433,33 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 		return err
 	}
 
-	// 4.1. read modeenv
+	// 4.1 verify that ubuntu-data comes from where we expect it to
+
+	// since ubuntu-data has already been mounted, see if there is an encrypted
+	// device that we should try to boot from
+	diskOpts := &disks.Options{}
+	_, err = disk.FindMatchingPartitionUUID("ubuntu-data-enc")
+	if err == nil {
```

Comment:
shouldn't we not ignore errors that are not finding the the partition with the name at all? see also later comment

Reply:
with your other suggestion to return whether it was found or not from `secboot.Unlock...` this ended up being immaterial actually